.circleci/config.yml

```diff
@@ -1,44 +1,297 @@
+    environment:
+      TESTING_PROFILE: confidential-workflow-tests
+    steps:
+      - setup_machine
+      - setup_for_tests
+      - setup_machine_env_vars
+      - setup_for_integration_tests
+      - run_tests
+      - clean_and_save_cache
+      - save_test_results
+      - send_coverage
+
+  confidential-tool-integration-tests:
+    # We must use machine executor instead of Docker executor and setup_remote_docker
+    # https://circleci.com/docs/2.0/executor-intro/#machine
+    # perhaps because cwltool cannot mount a volume from our job space
+    # into CircleCI remote Docker
+    # https://circleci.com/docs/2.0/building-docker-images/#mounting-folders
+    executor: machine_integration_test_exec
+    environment:
+      TESTING_PROFILE: confidential-tool-tests
+    steps:
+      - setup_machine
+      - setup_for_tests
+      - setup_machine_env_vars
+      - setup_for_integration_tests
+      - run_tests
+      - clean_and_save_cache
+      - save_test_results
+      - send_coverage
+
+
+    # This catagory seems to be covered by confidential and non-confidential tests
+    #integration-tests:
+    ## We must use machine executor instead of Docker executor and setup_remote_docker
+    ## https://circleci.com/docs/2.0/executor-intro/#machine
+    ## perhaps because cwltool cannot mount a volume from our job space
+    ## into CircleCI remote Docker
+    ## https://circleci.com/docs/2.0/building-docker-images/#mounting-folders
+    #executor: machine_integration_test_exec
+    #environment:
+    #  TESTING_PROFILE: integration-tests
+    #steps:
+    #  - setup_machine
+    #  - setup_for_tests
+    #  - setup_machine_env_vars
+    #  - setup_for_integration_tests
+    #  - run_tests
+    #  - clean_and_save_cache
+    #  - save_test_results
+    #  - send_coverage
+
+
+
+  build:
+    docker: # run the steps with Docker
+      - image: cimg/openjdk:11.0.8
+        environment:
+          # Java can read cgroup. Sadly the cgroup in
+          # CircleCI is wrong. Have to manually set. Nothing to do with surefire
+          # plugin, it has its own JVM. The two of these must add up to a bit less than 4GB.
+          JAVA_TOOL_OPTIONS: -Xmx512m
     steps: # a collection of executable commands
       - checkout # check out source code to working directory
       - restore_cache: # restore the saved cache after the first run or if `pom.xml` has changed
           # Read about caching dependencies: https://circleci.com/docs/2.0/caching/
-          key: dockstore-java-{{ checksum "pom.xml" }}
+          key: dockstore-cli-java-{{ checksum "pom.xml" }}
+      - install_yq
+      - install-git-secrets
+      - install_confidential_test_data
+      - run:
+          name: build
+          command: mvn -B clean install -DskipTests
+
+        # Running scan must occur after build
+      - run_git_secrets_scan
+      - check_license_file
+      - run:
+          name: copy m2 manually
+          command: cp -r ~/.m2 .
```

Comment:
There is already code that saves and restores the .m2 directory, e.g., line 196. Can you use that instead?

Reply:
Not sure I will have to investigate.

Reply:
I feel this is worth resolving -- we should only do it one way if possible.

Reply:
Now .m2 is only saved to cache (when calling save_cache)